src/xbase/xrpc/xrpc.go

```diff
@@ -49,26 +49,39 @@ func (s *Service) Start() error {
 		return errors.New("xrpc.Start.error[Please RegisterService first]")
 	}
 
-	lis, err := net.Listen("tcp", s.opts.ConnectionStr)
+	err := s.setListener()
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	s.listener = lis
 
 	go func() {
 		for {
-			conn, err := lis.Accept()
+			conn, err := s.listener.Accept()
 			if err != nil {
 				s.opts.Log.Error("xrpc.accept.error[%v]", err)
 				return
 			}
 			go s.server.ServeConn(conn)
 		}
 	}()
-	s.opts.Log.Warning("xrpc.Start.listening.on[%v]", lis.Addr())
+	s.opts.Log.Warning("xrpc.Start.listening.on[%v]", s.listener.Addr())
 	return nil
 }
 
+func (s *Service) setListener() error {
```

Comment:
Maybe we use a Time.Ticker here will be better ? e.g.:
 go func() {
       for range ticker {
               event1
               event2
               ...
}
}
select {
case event1:
...
case event2:
...
}

Reply:
This is not a scheduled task. The logic here is to set the listener. If announce listener fail, wait for 1s, and then try again. A total of 30 times. If more than 30 times, return error.
Only if announce listener success, the following steps can be carried out.